server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -8,132 +8,143 @@
 
 package org.opensearch.index.shard;
 
+import org.apache.lucene.index.SegmentInfos;
 import org.apache.lucene.store.Directory;
-import org.apache.lucene.store.IOContext;
-import org.opensearch.test.OpenSearchTestCase;
+import org.apache.lucene.store.FilterDirectory;
+import org.apache.lucene.tests.store.BaseDirectoryWrapper;
+import org.junit.After;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.concurrent.GatedCloseable;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.engine.InternalEngineFactory;
+import org.opensearch.index.store.RemoteSegmentStoreDirectory;
+import org.opensearch.index.store.Store;
 
 import java.io.IOException;
-import java.nio.file.NoSuchFileException;
+import java.util.Map;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.doThrow;
+public class RemoteStoreRefreshListenerTests extends IndexShardTestCase {
```

Comment:
Could you please add tests for RemoteStoreRefreshListener in case shard flips from primary to replicate or vise versa